elastic-workers-ss2.md

```diff
@@ -0,0 +1,74 @@
+# Elastic Workers for Spark in Oshinko
+
+## Introduction
+
+Describes modifications and enhancements to add elastic worker logic to Oshinko
+Spark clusters.  This solution is driven primarily by the information
+available from Spark's published metrics, and Spark cluster information
+stored by Oshinko itself.
+
+## Problem statement
+
+The purpose of this enhancement is to allow Spark clusters created in Oshinko
+to manage their own compute scale by either increasing or decreasing their
+number of Spark worker pods, based on the number of Executors being requested
+by Spark's dynamic Executor feature.
+
+## Proposed solution
+
+The proposed solution involves the following modifications:
+
+* Add containers and configuration to the Spark Master pod for Spark metric
+sinks.  In particular, carbon and graphite containers.   This modification
+is proceeding as a separate Spec and PR, and so the carbon/graphite work
+can be considered as a "blocker" for this elastic worker enhancement.
+* Add configuration settings to Spark to enable dynamic Executors.  This can
+be accomplished either by modifying the Spark Dockerfiles to add this
+configuration to `spark_defaults.conf`, or it can be done by adding the
+configurations to the `SPARK_MASTER_OPTS` and `SPARK_WORKER_OPTS` environment
+variables, which can be done via the template.
```

Comment:
In the case of oshinko-created clusters we probably would just programmatically add these confs to the environment for the pod.